src/circuits/circuits-util.tcc

```diff
@@ -102,6 +102,70 @@ libsnark::pb_variable_array<FieldT> from_bits(
     return acc;
 };
 
+// Sum 2 binary strings, discard last carry
+template<typename FieldT, size_t BitLen>
+std::array<FieldT, BitLen> binary_field_addition_no_carry(
```

Comment:
@AntoineRondelet We have similar functions (xor, addition...) working on _bits_ and not_fields_ in types/bits.tcc.
Would you agree to merge circuits-util.* into bits.* ?

Reply:
> Would you agree to merge circuits-util.* into bits.* ?

I'm not sure I understand what you mean @rrtoledo. The circuits-util.* files aim to provide util functions for the circuit and manipulate bits represented by FieldT as shown on the line you commented. The `bits.*` files on the other hand, are used to introduce new types and functions that are useful to manipulate them.

Reply:
The `circuit-utils` aim to be called inside the circuit while the others can be used outside

Reply:
Fair enough.

I did not see this distinction between the files, only that they both process bits (encoded as bool or FieldT).